### ISSUE TITLE
IntelliSence works inside of the script tag but doesn't work inside of <template>

### ISSUE THREAD

**Issue by TalkativeGuy** [vuejs/language-tools#4416]
https://github.com/vuejs/language-tools/assets/39724924/8859a403-4081-499c-8707-6931919ea761

## Description
As can be seen on the video, the pop-up hints for the data values work when typed inside of the script tag (e. g. in a method), but when it comes to the template - it only works for the "top layer", so the hints do display the data values (not always though), but not their properties after the dot. The issue applies for both anonymous an ts-typed objects. Doesn't display errors if something is wrong either.

## What was attempted
- Reinstalling the extension
- Enabling/disabling the built-in TS extension (```@builtin types```) and another single vue snippets extension I have
- Downgrading to 1.8.27 as was suggested in another similar issue. Surprisingly, it started acting the other way around: working in template, not working in script tag. LoL.
- Reloading VS Code for a million times including when downgraded to 1.8.27
- I wanted to try and reinstall VS Code completely, but then I remembered I just had it installed a couple of days ago on by partner's laptop as mine was unavailable for a while, and it had the same issue.

Back when it was Volar, it worked perfectly fine in both HTML *and* TS, but now it won't let me install it back again and my experience is now ruined :(

**Comment by so1ve** [comment]
Could you please provide your code? I'm not sure if it is https://github.com/vuejs/core/issues/9296

**Comment by TalkativeGuy** [comment]
> Could you please provide your code? I'm not sure if it is [vuejs/core#9296](https://github.com/vuejs/core/issues/9296)

Even in the most simple file possible it fails to recognize (it has the exactly same structure as the file in the video)

![image](https://github.com/vuejs/language-tools/assets/39724924/cf2a0f02-5c06-4a93-820d-0459c2ae82ca) ![image](https://github.com/vuejs/language-tools/assets/39724924/ba1e4062-6e3c-46c5-8ff0-eb2afab23af7)

I've figured out though that in my other project that runs on Vue CLI (instead of Vite for this one), everything works fine. So must be something about Vite? Here's my ```tsconfig.json```:
```json
{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": false,
    "module": "ESNext",
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "skipLibCheck": true,
    "experimentalDecorators": true, 
    "emitDecoratorMetadata": true,
    "types": ["node", "vite/client"],

    /* Bundler mode */
    "moduleResolution": "node",
    "allowImportingTsExtensions": true,
    // "resolveJsonModule": false,
    // "isolatedModules": false,
    "noEmit": true,
    "jsx": "preserve",

    /* Linting */
    "strict": true,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "noFallthroughCasesInSwitch": true,
	
    /* These ones I tried switching back and forth already */
    "importHelpers": true,
    "esModuleInterop": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
  },
  "include": ["src/**/*.ts", "src/**/*.tsx", "src/**/*.vue, ./src/**/**/*.ts"],
  "references": [{ "path": "./tsconfig.node.json" }],
}
```

And ```vite.config.json```:

```json
import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [vue()],
  server: {
    port: 3000
  }
})
```

**Comment by MrMarble** [comment]
I have the same error, even the type definition shows `any` instead of the correct types.

Works fine inside `script` tag
![imagen](https://github.com/vuejs/language-tools/assets/4268580/95087ea4-223b-44e0-b330-946047c10e6a)

Nothing inside `template`
![imagen](https://github.com/vuejs/language-tools/assets/4268580/846933f9-b0da-4d30-93ff-a7895a542631)


- vue: `v3.4.27`
- vite: `v5.2.11`
- vscode extension: `v2.0.19`
- vscode: `1.89.1`

**Comment by TalkativeGuy** [comment]
I've figured it out. Turns out when trying to solve the problem I made a typo in the ```include``` section in ```tsconfig.json```. Looks like I have solved the problem after that by doing something else but it kept not working because of that typo. I think updating all the modules did the job but of course can't say for sure now. Anyway if anyone runs into a problem like this try doing everything I listed in the question. Except for downgrading to a previous version as this one seems to work fine.

The typo I had just in case you're interested:
```json
"include": ["src/**/*.ts", "src/**/*.tsx", "src/**/*.vue, ./src/**/**/*.ts"],
```
Instead of
```json
"include": ["src/**/*.ts", "src/**/*.tsx", "src/**/*.vue", "./src/**/**/*.ts"],
```
